object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/parser/ExpressionBuilder.java

```diff
@@ -106,6 +106,8 @@ public void addNestedComparison(PropertyPath<?> propertyPath, ComparisonExpr.Typ
       if (!nestedExprMap.containsKey(joinAlias)) {
          nestedExprMap.put(joinAlias, new NestedExpr(getNestedPath(propertyPath)));
          push(nestedExprMap.get(joinAlias));
+      }else if(stack.peek() instanceof LazyNegationExpr && !nestedExprMap.isEmpty()){
```

Comment:
the second check seems to be redundant

```suggestion
      } else if (stack.peek() instanceof LazyNegationExpr) {
```

Reply:
hm you're right, I will fix these soon. Thanks for the feedback!